lib/std/private/underscored_calls.nim

```diff
@@ -41,9 +47,22 @@ proc underscoredCalls*(result, calls, arg0: NimNode) =
         result.add underscoredCall(a, arg0)
 
   expectKind calls, nnkArgList
+
+  # handle the case when arg0 is x.dup
```

Comment:
I don't understand why this case has to be handled with special logic. I don't understand how ``arg0[^1].kind == nnkSym`` indicates ``isDup``.

Reply:
Because the code ``x.dup(sort).with`` will make the arg0 like this:
```nim
var dupResult = x
sort(dupResult)
dupResult
```
Of cause, it is tricky, not understandable and depends on implementation of ``dup``, I will remove it later.